src/components/pie-stimulus-layout/pie-stimulus-layout.tsx

```diff
@@ -11,24 +11,81 @@ export class PieStimulusLayout {
       el.style.height = `calc(100vh - ${boundRect.top}px`;
     }
   }
+  private resizer: HTMLDivElement;
+
+  // shows us if resizing is activated or not
+  @Prop() allowedResize: boolean = false;
+
+  @State() isResizing = false;
+  @State() initialX = 0;
+  @State() initialLeftFlex = 0.5;
+  @State() initialRightFlex = 0.5;
 
   componentDidRender() {
     PieStimulusLayout.handleElements();
   }
 
   componentDidLoad() {
     PieStimulusLayout.handleElements();
+    if(this.allowedResize){
+      this.resizer.addEventListener('mousedown', (e) => {
+        this.isResizing = true;
+        this.initialX = e.clientX;
+      });
+      window.addEventListener('mousemove', this.handleMouseMove.bind(this));
+      window.addEventListener('mouseup', this.handleMouseUp.bind(this));
+    }
   }
 
   componentDidUpdate() {
     PieStimulusLayout.handleElements();
   }
 
+  componentDidUnload() {
+    if(this.allowedResize){
+      window.removeEventListener('mousemove', this.handleMouseMove.bind(this));
+      window.removeEventListener('mouseup', this.handleMouseUp.bind(this));
+    }
+  }
+
+  private handleMouseMove(e: MouseEvent) {
+    if (this.isResizing && this.allowedResize) {
+      const deltaX = e.clientX - this.initialX;
+      const containerWidth = this.resizer.parentElement.clientWidth;
+
+      const newLeftFlex = (containerWidth * this.initialLeftFlex + deltaX) / containerWidth;
+      const newRightFlex = 1 - newLeftFlex;
+      if (newLeftFlex >= 0 && newRightFlex >= 0 && newLeftFlex <= 1 && newRightFlex <= 1) {
+        this.initialLeftFlex = newLeftFlex;
+        this.initialRightFlex = newRightFlex;
+
+        // Update the initial cursor position
+        this.initialX = e.clientX;
+      }
+    }
+  }
+
+  private handleMouseUp() {
+    this.isResizing = false;
+  }
+
+
   render() {
     return (
       <div id="pie-stimulus-container">
-        <div id="stimulus"><slot name='stimulus' /></div>
-        <div id="item"><slot name='item' /></div>
+        <div id="stimulus"
```

Comment:
if we'd define the style above and make it {} if allowedResize is false, would it work?